src/GregClient/GregClient.cs

```diff
@@ -40,7 +40,15 @@ private IRestResponse ExecuteInternal(Request m)
 
             if (m.RequiresAuthorization)
             {
-                AuthProvider.SignRequest(ref req, _client);
+                var reqToSign = new RestRequest(req.Resource, req.Method);
```

Comment:
* can you add comments for what this actually does?
Am I reading it right that it signs the old request and builds a new one - and then transfers some info from old to new, and then drops the old one?

Reply:
It builds a subset of the current request, signs that subset, and then copies the signed data back to the current request.